### ISSUE TITLE
Problem with previous epoch balance

### ISSUE THREAD

**Issue by mkalinin** [ethereum/eth2.0-specs#243]
### Issue
`previous_epoch_boundary_attesting_balance` calculated for epoch with `N` boundary may not be equal to `this_epoch_boundary_attesting_balance` with `N - EPOCH_LENGTH` boundary. 

The former calculation uses validator set that has been affected by rewards and registry changes which have happened _after_ `this_epoch_boundary_attesting_balance` calculation took place.

As a result there are two different attesting balances for the same epoch, depending on slot.

### Proposed solution
It could probably be solved by storing original balance in the state. Something like
```
this_epoch_boundary_attesting_balance = sum([get_effective_balance(v) for v in this_epoch_boundary_attesters])
previous_epoch_boundary_attesting_balance = state.latest_attesting_balance

...

# At the end of epoch transition
state.latest_attesting_balance = this_epoch_boundary_attesting_balance
```

**Comment by vbuterin** [comment]
The other solution is to mark this as an asterisk and explicitly not worry about it, because there's a pretty low bound on the extent to which validator balances can change epoch-by-epoch so it affects consensus safety by at most a fraction of a percentage point.

**Comment by mkalinin** [comment]
Yep, actually, I was thinking about the same. Probably the biggest difference between balances would be in case of temporal eclipse when there've been only one block proposed during whole epoch and validators got penalized. But even in this case the difference would be negligible.

**Comment by hwwhww** [comment]
@djrtwo @mkalinin 
Is this issue fixed?
